improver/calibration/dataframe_utilities.py

```diff
@@ -510,21 +602,40 @@ def truth_dataframe_to_cube(df: DataFrame, training_dates: DatetimeIndex,) -> Cu
         time_coord = _define_time_coord(adate, time_bounds)
         height_coord = _define_height_coord(time_df["height"].values[0])
 
-        cube = build_spotdata_cube(
-            time_df["ob_value"].astype(np.float32),
-            time_df["cf_name"].values[0],
-            time_df["units"].values[0],
-            time_df["altitude"].astype(np.float32),
-            time_df["latitude"].astype(np.float32),
-            time_df["longitude"].astype(np.float32),
-            time_df["wmo_id"].values.astype("U5"),
-            scalar_coords=[time_coord, height_coord],
-        )
-        cubelist.append(cube)
+        if var_type == "threshold":
+            for var_val in sorted(df[var_type].unique()):
+                var_coord = DimCoord(np.float32(var_val), long_name=var_type, units=1)
+                var_df = time_df.loc[time_df[var_type] == var_val]
+                cube = build_spotdata_cube(
+                    (var_df["ob_value"] > var_val).astype(np.int32),
+                    var_df["cf_name"].values[0],
+                    var_df["units"].values[0],
+                    var_df["altitude"].astype(np.float32),
+                    var_df["latitude"].astype(np.float32),
+                    var_df["longitude"].astype(np.float32),
+                    var_df["wmo_id"].values.astype("U5"),
+                    scalar_coords=[time_coord, height_coord, var_coord],
+                )
+                cubelist.append(cube)
+        else:
+            cube = build_spotdata_cube(
+                time_df["ob_value"].astype(np.float32),
+                time_df["cf_name"].values[0],
+                time_df["units"].values[0],
+                time_df["altitude"].astype(np.float32),
+                time_df["latitude"].astype(np.float32),
+                time_df["longitude"].astype(np.float32),
+                time_df["wmo_id"].values.astype("U5"),
+                scalar_coords=[time_coord, height_coord],
+            )
+            cubelist.append(cube)
 
     if not cubelist:
         return
-    return cubelist.merge_cube()
+
+    cube = cubelist.merge_cube()
+    cube.attributes["cube_type"] = "observations"
```

Comment:
I don't think we should be setting forecasts to "observations" cube_type.  We need to check this